frontend/src/app/App.tsx

```diff
@@ -27,12 +35,38 @@ const App: React.FC = () => {
     setIsNavOpen(!isNavOpen);
   };
 
+  // when there is a segment key and username and cluster id, initialize segment.io script
+  React.useEffect(() => {
+    if (segmentKey && loaded && !loadError && username && clusterID) {
+      window.clusterID = clusterID;
+      initSegment({ segmentKey, username });
```

Comment:
Could we have `initSegment` return the `analytics` and store with redux rather than using `window`?

Reply:
@jeff-phillips-18 If we use redux, will it be difficult to be called? We cannot use redux hook functions inside the `fireTrackingEvent` function, so every time we call it we need to pass it through the arguments. Do you have any suggestions about how to avoid this situation if we want to use redux?

Reply:
I think it is ok to pass the analytics to the `fireTrackingEvent` function.